// Protocol buffer specification for evaluation of a model.
// WIP: DO NOT USE.

syntax = "proto2";

import "tagset/coarse_pos/coarse_tag_enum.proto";
import "tagset/dep_labels/dep_label_enum.proto";
import "tagset/fine_pos/fine_tag_enum.proto";

package model;

message Evaluation {
  // arc accuracy of the model.
  optional float arc_accuracy = 1;
  // unlabeled attachment score with this model.
  optional float uas_total = 2;
  // labeled attachment score with this model.
  optional float las_total = 3;
  // typed unlabeled attachment score with this model.
  optional TypedUas typed_uas = 4;
  // typed labeled attachment precision with this model.
  optional TypedLasPrec typed_las_prec = 5;
  // typed labeled attachment recall with this model.
  optional TypedLasRecall typed_las_recall = 6;
  // typed labeled attachment f1 with this model.
  optional TypedLasF1 typed_las_f1 = 7;
}

message TypedUas {
  repeated Analysis uas = 1;
}

message TypedLasPrec {
  repeated Analysis prec = 1;
}

message TypedLasRecall {
  repeated Analysis recall = 1;
}

message TypedLasF1 {
  repeated LasF1 f1 = 1;
}

message Analysis {
  optional tagset.dep_labels.Tag label = 1;     /*actually required*/
  optional float score = 2;                     /*actually required*/
}

message LasF1 {
  optional tagset.dep_labels.Tag label = 1;   /*actually required*/
  optional int32 count = 2;                   /*actually required*/
  optional float prec = 3;                    /*actually required*/
  optional float recall = 4;                  /*actually required*/
  optional float f1  = 5;                     /*actually required*/
}